csrc/ir_nodes.cpp

```diff
@@ -2500,6 +2500,33 @@ void IterDomain::parallelize(ParallelType t) {
     return;
   }
 
+  // Leaf domains are domains that are not used by any other domains.
+  auto isLeafDomain = [&]() {
+    for (auto expr : uses()) {
+      if (expr->isA<Split>() || expr->isA<Merge>() || expr->isA<Resize>()) {
+        return false;
+      }
+    }
+    return true;
+  };
```

Comment:
Can this just be `expr->uses().empty()`?

Reply:
I started with `expr->uses().empty()` then there are some other uses in the lowering pass. I rechecked and it works fines with just `expr->uses().empty()` . The previous use is probabaly for the serial iter domain in the loop iteration when lowering.